mono/metadata/threads.c

```diff
@@ -2096,16 +2079,15 @@ ves_icall_System_Threading_WaitHandle_Wait_internal (gpointer *handles, gint32 n
 		if (ret != MONO_W32HANDLE_WAIT_RET_ALERTED)
 			break;
 
-		exc = mono_thread_execute_interruption_ptr ();
-		if (exc) {
-			mono_error_set_exception_instance (error, exc);
+		// Only ever create one handle in the loop.
+		MonoExceptionHandle exc;
+		if (mono_thread_execute_interruption (&exc)) {
+			mono_error_set_exception_handle (error, exc);
```

Comment:
But this is still assigning to `exc` a handle that's allocated (and deallocated) within the body of `mono_thread_execute_interruption` and then expecting to get a non-garbage handle in `exc`.

Just declaring a handle-typed variable does not allocate a handle.

Reply:
Of course, because there are no constructors..